searx/engines/google.py

```diff
@@ -1,210 +1,200 @@
-map_near_title = './/h4'
-map_near_url = './/h4/a/@href'
-map_near_phone = './/span[@class="nobr"]'
-
-# images
-images_xpath = './/div/a'
-image_url_xpath = './@href'
-image_img_src_xpath = './img/@src'
-
-# property names
-# FIXME : no translation
-property_address = "Address"
-property_phone = "Phone number"
-
-
-# remove google-specific tracking-url
-def parse_url(url_string, google_hostname):
-    # sanity check
-    if url_string is None:
-        return url_string
-
-    # normal case
-    parsed_url = urlparse(url_string)
-    if (parsed_url.netloc in [google_hostname, '']
-            and parsed_url.path == redirect_path):
-        query = dict(parse_qsl(parsed_url.query))
-        return query['q']
-    else:
-        return url_string
+# ------------------------
+
+# google results are grouped into <div class="g" ../>
+results_xpath = '//div[@class="g"]'
+
+# google *sections* are no usual *results*, we ignore them
+g_section_with_header='./g-section-with-header'
+
+# the title is a h3 tag relative to the result group
+title_xpath = './/h3[1]'
 
+# in the result group there is <div class="r" ../> it's first child is a <a
+# href=...> (on some results, the <a> is the first "descendant", not ""child")
+href_xpath = './/div[@class="r"]//a/@href'
+
+# in the result group there is <div class="s" ../> containing he *content*
+content_xpath = './/div[@class="s"]'
+
+# Suggestions are links placed in a *card-section*, we extract only the text
+# from the links not the links itself.
+suggestion_xpath = '//div[contains(@class, "card-section")]//a'
+
+# Since google does *auto-correction* on the first query these are not really
+# *spelling suggestions*, we use them anyway.
+spelling_suggestion_xpath = '//div[@class="med"]/p/a'
 
-# returns extract_text on the first result selected by the xpath or None
 def extract_text_from_dom(result, xpath):
+    """returns extract_text on the first result selected by the xpath or None"""
     r = eval_xpath(result, xpath)
     if len(r) > 0:
         return extract_text(r[0])
     return None
 
-
-# do search-request
-def request(query, params):
-    offset = (params['pageno'] - 1) * 10
-
-    if params['language'] == 'all' or params['language'] == 'en-US':
-        language = 'en-GB'
-    else:
-        language = match_language(params['language'], supported_languages, language_aliases)
+def get_lang_country(params, lang_list, custom_aliases):
+    """Returns a tuple with *langauage* on its first and *country* on its second
+    position."""
+    language = params['language']
+    if language == 'all':
+        language = 'en-US'
 
     language_array = language.split('-')
-    if params['language'].find('-') > 0:
-        country = params['language'].split('-')[1]
-    elif len(language_array) == 2:
+
+    if len(language_array) == 2:
         country = language_array[1]
     else:
-        country = 'US'
-
-    url_lang = 'lang_' + language
+        country = language_array[0].upper()
```

Comment:
Just a comment, when the search language is 'English - en':
* ```Accept-Language: en-EN,en;q=0.8,en;q=0.5```
* ``` /search?q=itme&hl=en-EN&lr=lang_en&ie=utf8&oe=utf8&start=0```
 
But ```en-EN``` works seems okay.

Reply:
Thanks! Applied: 20bbed1d